### PR TITLE
storagenode/pieces: Add EmptyTrash functionality

### PR DESCRIPTION
**Pull request by isaachess**
What: Add EmptyTrash functionality and chore to periodically run to empty the trash after 7 days. The chore runs once per day and deletes anything that has been in the trash for 7 days or longer.

Why: We need a way to empty the trash.

Please describe the tests:
 - Test 1: `TestTrashAndRestore` Has been updated to test for this new functionality.
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: storagenode/pieces/store.go [hunk 1]
```diff
@@ -312,6 +312,27 @@ func (store *Store) Trash(ctx context.Context, satellite storj.NodeID, pieceID s
...
+	defer mon.Task()(&ctx)(&err)
+
+	deletedIDs, err := store.blobs.EmptyTrash(ctx, satelliteID[:], trashedBefore)
+	if err != nil {
+		return Error.Wrap(err)
+	}
+
+	var pieceID storj.PieceID
+	for _, deletedID := range deletedIDs {
+		pieceID, err = storj.PieceIDFromBytes(deletedID)
```
Comment (zeebo) [inline storagenode/pieces/store.go]: This is overwriting err to nil every time, which misses the collected `DeleteExpiration` errors.

File: storagenode/pieces/store_test.go [hunk 2]
```diff
@@ -237,6 +244,29 @@ func TestTrashAndRestore(t *testing.T) {
...
+					},
+				},
+				{
+					pieceID:    testrand.PieceID(),
+					expiration: time.Now().Add(24 * time.Hour),
+					trashDur:   trashDurToBeKept,
+					files: []testfile{
+						{
+							data:      testrand.Bytes(size),
+							formatVer: filestore.FormatV1,
```
Comment (zeebo) [inline storagenode/pieces/store_test.go]: how is this piece different from the previous one? should one of them be FormatV0?

Comment (isaachess) [reply storagenode/pieces/store_test.go]: It's not, I just wanted to prove it works with a bunch of pieces. It has a different pieceID, and different data, so it is a different piece, but it is also a v1 piece.

File: storagenode/pieces/store.go (hunk 1 above)
Comment (isaachess) [reply storagenode/pieces/store.go]: Done.

File: storagenode/peer.go [hunk 3]
```diff
@@ -248,6 +250,14 @@ func New(log *zap.Logger, full *identity.FullIdentity, db DB, revocationDB exten
 			peer.DB.PieceSpaceUsedDB(),
 		)
 
+		peer.Storage2.TrashChore = pieces.NewTrashChore(
+			log.Named("pieces:trashchore"),
+			24*time.Hour,   // choreInterval: how often to run the chore
+			7*24*time.Hour, // trashExpiryInterval: when items in the trash should be deleted
```
Comment (simongui) [inline storagenode/peer.go]: Should these options be configurable? PR looks good to me! I don't think a merge should be pending on the answer of this question :)

**Pull request merged**